_posts/2017-04-27-Rust-1.17.md

```diff
@@ -0,0 +1,352 @@
+             at /checkout/src/libstd/panicking.rs:371
+   4: std::panicking::rust_panic_with_hook
+             at /checkout/src/libstd/panicking.rs:549
+   5: std::panicking::begin_panic
+   6: foo::main
+   7: __rust_maybe_catch_panic
+             at /checkout/src/libpanic_unwind/lib.rs:98
+   8: std::rt::lang_start
+             at /checkout/src/libstd/panicking.rs:433
+             at /checkout/src/libstd/panic.rs:361
+             at /checkout/src/libstd/rt.rs:57
+   9: main
+  10: __libc_start_main
+  11: _start
+```
+
+By default. As the message says, you can set an environment variable to get the full
+backtrace. We may be able to do more cleanup in the future; see [this bug](https://github.com/rust-lang/rust/pull/40264) for more.
+
+See the [detailed release notes][notes] for more.
+
+#### Cargo features
+
+Other than the previously mentioned `build.rs` changes, Cargo has a few new improvements.
+[`cargo check --all`](https://github.com/rust-lang/cargo/pull/3731) and
+[`cargo run --package`](https://github.com/rust-lang/cargo/pull/3691) are two missing
+flags that are now supported.
+
+You can now [opt in to ignoring SSL revocation checks](https://github.com/rust-lang/cargo/pull/3699). The default is still to check, of course.
+
+A new field in `Cargo.toml`, `required-features`, lets you [specify specific features
+that must be set for a target to be built](https://github.com/rust-lang/cargo/pull/3667).
+Here's an example: let's say that we are writing a crate that interacts with databases,
+and that we support multiple databases. We might have this in our `Cargo.toml`:
+
+```toml
+[features]
+# ...
+postgres = []
+sqlite = []
+tools = []
+```
+
+The `tools` feature allows us to include extra tooling, and the `postgres` and `sqlite`
+features control which databses we want to support.
+
+Previously, `cargo build` would attempt to build all targets, which is normally what
+you want. But what if we had a `src/bin/postgres-tool.rs`, that would only really
+be relevant if the `postgres` and `tools` features would be enabled? Previously,
+we would have to write something like this:
+
+```rust
+#[cfg(not(all(feature = "postgres", feature = "tools")))]
+fn main() {
+    println!("This tool requires the `postgres` and `tools` features to be enabled.");
+}
+
+#[cfg(all(feature = "postgres", feature = "tools"))]
+fn main() {
+    // real code
+}
+```
+
+This is a lot of boilerplate to work around `cargo build`'s behavior. It's even
+more unfortunate with `examples/`, which are supposed to show off how to use
+your library, but this shenanigans is only relevant within the package, not if
+you were to try to use the example on your own.
+
+With the new `required-features` key, we can add this:
+
+```toml
+[[bin]]
+# ...
+required-features = ["postgres", "tools"]
+```
+
+Now, `cargo build` will only build our `postgres-tool` if we have the two features
+set, and so we can write a normal `fn main` without all the `cfg` nonsense getting
+in the way.
```

Comment:
> Can you fill in at least enough of the part currently marked # ... after [[bin]] to show how one of these targets works.

That would depend on the details of the project itself, which isn't the important part here.

> Maybe someone can ping whoever is in charge of those?

Guess who that is? :) Anyone could port this back to Cargo's docs, and in general, I have a "rehaul Cargo's docs) on my plate in the future.